src/eradius_server.erl

```diff
@@ -204,39 +206,36 @@ do_radius(ServerPid, ReqKey, Handler = {HandlerMod, _}, NasProp, {udp, Socket, F
             TS2 = eradius_metrics:timestamp(milli_seconds),
             inc_counter(Cmds, NasProp, TS2 - TS1),
             gen_udp:send(Socket, FromIP, FromPort, EncReply),
-            case application:get_env(eradius, resend_timeout, 2000) of
-                0 -> 
-                    ServerPid ! {free, ReqKey};
-                ResendTimeout ->
-                    ServerPid ! {replied, ReqKey, self()},
-                    wait_resend_init(ServerPid, ReqKey, FromIP, FromPort, EncReply, ResendTimeout, ?RESEND_RETRIES)
+            ResendTimeout = application:get_env(eradius, resend_timeout, 2000),
```

Comment:
very small style comment, does not have to be implemented....

The case made IMHO more sense, the value of ResendTimeout is not used outside the if/case, so assigning it unconditionally is not really needed. I would have written this as:

           case application:get_env(eradius, resend_timeout, 2000) of
                ResendTimeout when is_integer(ResendTimeout), ResendTimeout > 0 ->
                    ServerPid ! {replied, ReqKey, self()},
                    wait_resend_init(ServerPid, ReqKey, FromIP, FromPort, EncReply, ResendTimeout, ?RESEND_RETRIES);
                _ -> 
                    ok
           end,